src/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.cs

```diff
@@ -1391,19 +1363,19 @@ public MethodBuilder DefineMethod(string name, MethodAttributes attributes, Call
             if (!m_isHiddenGlobalType)
             {
                 //If this method is declared to be a constructor, increment our constructor count.
-                if ((method.Attributes & MethodAttributes.SpecialName) != 0 && method.Name.Equals(ConstructorInfo.ConstructorName))
+                if ((method.Attributes & MethodAttributes.SpecialName) != 0 && method.Name!.Equals(ConstructorInfo.ConstructorName))
                 {
                     m_constructorCount++;
                 }
             }
 
-            m_listMethods.Add(method);
+            m_listMethods?.Add(method);
```

Comment:
This is adding a null check.  Why is it needed / in what situation is m_listMethods null here?  If it actually can be, it would seem the fix isn't to use `?.` but to instead ensure it's properly initialized.

Reply:
it wasn't needed, seems i marked it nullable at first then make it non nullable and this one left ...